quinn-h3/src/lib.rs

```diff
@@ -37,4 +39,206 @@ impl StreamType {
     pub fn encode<W: BufMut>(&self, buf: &mut W) {
         buf.write_var(self.0);
     }
+    pub fn decode<T: Buf>(buf: &mut T) -> Result<Self, UnexpectedEnd> {
+        Ok(StreamType(buf.get_var()?))
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
Probably better to put all the error types in a dedicated module rather than in `lib.rs`.